content/en/docs/concepts/storage/volumes.md

```diff
@@ -11,228 +11,193 @@ weight: 10
 
-#### azureFile CSI migration
+[FlexVolume](#flexvolume) is an another mechanism for storage plugins. Unlike CSI, FlexVolume is
+more closely tied to Kubernetes and its storage implementation. FlexVolume was developed before CSI.
 
-{{< feature-state for_k8s_version="v1.21" state="beta" >}}
+Storage plugin interfaces:
 
-The `CSIMigration` feature for `azureFile`, when enabled, redirects all plugin operations
-from the existing in-tree plugin to the `file.csi.azure.com` Container
-Storage Interface (CSI) Driver. In order to use this feature, the [Azure File CSI
-Driver](https://github.com/kubernetes-sigs/azurefile-csi-driver)
-must be installed on the cluster and the `CSIMigration` and `CSIMigrationAzureFile`
-[feature gates](/docs/reference/command-line-tools-reference/feature-gates/) must be enabled.
+   * [CSI](#csi)
+   * [FlexVolume](#flexvolume)
+   * [persistentVolumeClaim](#persistentvolumeclaim)
 
-Azure File CSI driver does not support using same volume with different fsgroups, if Azurefile CSI migration is enabled, using same volume with different fsgroups won't be supported at all.
 
-### cephfs
+### External Volume Types
 
-A `cephfs` volume allows an existing CephFS volume to be
-mounted into your Pod. Unlike `emptyDir`, which is erased when a pod is
-removed, the contents of a `cephfs` volume are preserved and the volume is merely
-unmounted. This means that a `cephfs` volume can be pre-populated with data, and
-that data can be shared between pods. The `cephfs` volume can be mounted by multiple
-writers simultaneously.
+Kubernetes directly implements some volume types for communicating with externally managed volumes, such as AWS EBS or Azure Disk Storage. These volume types are described as in-tree storage plugins. 
 
-{{< note >}}
-You must have your own Ceph server running with the share exported before you can use it.
-{{< /note >}}
+The CSI was introduced to standardize connectivity to externally managed volumes, such as cloud platform storage. However, Kubernetes still includes many volume types for connecting to specific cloud storage providers. 
 
-See the [CephFS example](https://github.com/kubernetes/examples/tree/{{< param "githubbranch" >}}/volumes/cephfs/) for more details.
+External Volume Types:
+   * [awsElasticBlockStore](#awselasticblockstore)
+   * [azureDisk](#azuredisk)
+   * [azureFile](#azurefile)
+   * [cinder](#cinder)
+   * [flocker](#flocker)
+   * [gcePersistentDisk](#gcepersistentdisk)
+   * [portworxVolume](#portworxvolume)
+   * [quobyte](#quobyte)
+   * [storageos](#storageos)
+   * [scaleIO](#scaleio)
+   * [vsphereVolume](#vspherevolume)
 
-### cinder
+## Specify subPaths on volumes {#using-subpath}
 
-{{< note >}}
-Kubernetes must be configured with the OpenStack cloud provider.
-{{< /note >}}
+The `volumeMounts.subPath` property specifies a sub-path inside the referenced volume instead of its root. For example, this property may be used to to share one volume for multiple uses in a single pod.
```

Comment:
"The `volumeMounts.subPath` property mounts the subdirectory of the volume into the container"